packages/core-mobile/app/new/features/track/screens/TrackTokenDetailScreen.tsx

```diff
@@ -178,13 +179,16 @@ const TrackTokenDetailScreen = (): JSX.Element => {
     }
   }, [openUrl, tokenInfo?.urlHostname, back])
 
-  const handleBuy = useCallback((): void => {
-    navigate({
-      // @ts-ignore TODO: make routes typesafe
-      pathname: '/buy',
-      params: { showAvaxWarning: 'true' }
-    })
-  }, [navigate])
+  const handleBuy = useCallback(
+    (contractAddress?: string): void => {
+      if (contractAddress === undefined) return
+      navigateToBuy({
+        showAvaxWarning: true,
+        tokenOrAddress: contractAddress
+      })
```

Comment:
actually I think it will be cleaner to have separate fields for token and address